CONTRIBUTING.md

```diff
@@ -61,6 +61,39 @@ This basically builds `terracognita` with the current sources.
+
+**Provider**
+
+All providers have to follow the [provider.Provider](https://github.com/cycloidio/terracognita/blob/master/provider/provider.go#L14) interface in order to be able to be integrated to the codebase. The best way would be to follow any of the other providers and modify it accordingly to your needs. Also reading the documentation of the interface will help a lot in any question regarding what means any of the functions.
+
+New providers have to be on the root with the package name being the name of the provider on Terraform (ex: Amazon is `aws`)
+
+The `NewProvider` will need all the information required to initialize the specific provider client(sdk), and the specific provider configuration so then Terraform can use it to read information remotely and load it to the Schemas. All the information has to be sent via parameters, no ENV variables.
+
+**Resources**
+
+Resources are each of one of the Terraform resources that this specific provider has and can import using TerraCognita. The main idea is that the `provider.Provider.Resources` returns all the instances of one resource. That information has to be fetched from the provider API using the sdk initialized before and for each instance we have to build the Terraform resource ID and return the resource. This then will be used by Terraform to read all the needed data from the provider to load the resource Schema (no need to do anything as this is abstracted into the caller of that function).
+
+To know what's the ID expected by Terraform you can check on the Terraform documentation of the resource ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#import)[TODO: link docs and tf code], sometimes is more complex and it needs several elements to build an ID ([doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy#import) and [code](https://github.com/hashicorp/terraform-provider-aws/blob/main/aws/resource_aws_iam_role_policy.go#L171)) so we have to set it on the right format for then be read for Terraform.
+
+As providers have a lot of resources we have created custom sdks on top of the provider sdks that we use to simplify things. Those wrappers are autogenerated from a configuration that is custom for each Provider. We do this so when we have to add new resources is easy, one of the main reasons is to also avoid pagination and always return all the resources directly. For example in `aws` we have the [configuration](https://github.com/cycloidio/terracognita/blob/master/aws/cmd/functions.go) which has the [go:generate](https://github.com/cycloidio/terracognita/blob/master/aws/reader/connector.go#L56) that generates the end [reader](https://github.com/cycloidio/terracognita/blob/master/aws/reader/reader.go). We have similar things for the other sdks
```

Comment:
> We do this so when we have to add new resources is easy, one of the main reasons is to also avoid pagination and always return all the resources directly.

We do this to make adding new resources easy; by automating the process and always returning all elements - thus avoiding repetitive code writing and pagination logic.

> We have similar things for the other sdks

We have similar logic for the other SDKs*